commons-test/src/main/java/org/infinispan/commons/test/PolarionTestNameListener.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.commons.test;
+
+import java.lang.reflect.Field;
+
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.internal.TestResult;
+
+/**
+ *
+ * This listener fixes the method name by overwriting the private `TestResult.m_name` field and performs some
+ * additional processing to ensure the resulting test name is unique.
+ *
+ * @author <a href='mailto:afield[at]redhat[dot]com'>Alan Field</a>
+ * @author <a href='mailto:dlovison[at]redhat[dot]com'>Diego Lovison</a>
+ */
+public class PolarionTestNameListener extends TestNGNameTranslator implements ITestListener {
```

Comment:
Shouldn't you remove `Polarion` from the name here as well, if you did it for `PolarionJUnitXMLWriter`?

Also seems weird to extend `TestNGNameTranslator` just to invoke a method, when you could have made the method static and invoked it directly.

Reply:
The class was renamed to TestNGTestNameListener. Created a static method